### PR TITLE
AKS: No longer rely on selflink of pod

### LINKED ISSUE #486: Pod metadata selflink stopped working in Azure AkS and GKE
Solution: Manually create URL instead of relying on selfLink

### PR DESCRIPTION
**Pull request by skarsaune**
# Changes

- :bug: Handle changes in AKS and GKE

/kind bug

Fixes #486

**Release Note**

```release-note

Fixes bug that made it impossible to connect to pods in later versions of AKS and GKE
```

### CODE REVIEW

File: client/kubernetes/src/main/java/org/jolokia/kubernetes/client/KubernetesJmxConnector.java [hunk 1]
```diff
@@ -118,15 +119,13 @@ protected J4pClient expandAndProbeUrl(KubernetesClient client,
...
+  public static StringBuilder buildProxyPath(Pod pod, String protocol, String port, String path) {
+    final ObjectMeta metadata = pod.getMetadata();
+    final StringBuilder url = new StringBuilder("/api/").append(pod.getApiVersion()).append("/namespaces/").append(metadata.getNamespace()).append("/pods/");
     if (protocol != null && !protocol.equals("http:")) {
-      //if we need to specify protocol, inject in url
-      final int index = url.lastIndexOf(pod.getMetadata().getName());
-      if (index > -1) {
-        url.insert(index, protocol);
-      }
+      url.append(protocol);
```
Comment (rhuss) [inline client/kubernetes/src/main/java/org/jolokia/kubernetes/client/KubernetesJmxConnector.java:126]: Haven't seen the protocol in API URLs really before. Why do you need it here ?

Comment (skarsaune) [reply client/kubernetes/src/main/java/org/jolokia/kubernetes/client/KubernetesJmxConnector.java:126]: It allows you to specify https instead of http

**Comment by skarsaune** [comment]
The self link is not part of the API even in docker-desktop now, so even more urgent now

**Comment by rhuss** [comment]
thanks!

**Pull request merged**